addons/account/models/account_account.py

```diff
@@ -976,9 +976,16 @@ def _adapt_parent_account_group(self, company=None):
               ORDER BY child.id, char_length(parent.code_prefix_start) DESC
             )
             UPDATE account_group child
-               SET parent_id = relation.parent_id
+               SET parent_id = CASE
+                                  WHEN child.id = relation.parent_id THEN NULL
+                                  ELSE relation.parent_id
+                               END
               FROM relation
-             WHERE child.id = relation.child_id
+             WHERE (
+                    child.id = relation.parent_id
+                    OR
+                    child.id = relation.child_id
+                    )
```

Comment:
I don't get why you need something that complex.
Won't you possibly update multiple times the same rows?